.github/workflows/integration_test.yaml

```diff
@@ -159,152 +159,46 @@ jobs:
       working-directory: ${{ inputs.working-directory }}
   integration-test:
     name: Integration tests
-    strategy:
-      matrix:
-        series: ${{ fromJSON(inputs.series) }}
-        modules: ${{ fromJSON(inputs.modules) }}
-        ${{ insert }}: ${{ fromJSON(inputs.extra-test-matrix) }}
-      fail-fast: false
-    runs-on: ${{ needs.get-runner-image.outputs.runs-on }}
+    uses: ./.github/workflows/integration_test_run.yaml
```

Comment:
There is some limit which GitHub imposes for how deeply workflows can be nested, I think it is 4. I believe that the test and publish charm refers to this workflow which means that it would already require a depth of 3 which means that we are getting close to the limit

Reply:
Here is the documentation: https://docs.github.com/en/actions/using-workflows/reusing-workflows#nesting-reusable-workflows

Reply:
Yes, that's right. We are.